android/src/main/java/yeniellandestoy/native_shared_preferences/MethodCallHandlerImpl.java

```diff
@@ -226,20 +233,23 @@ private Map<String, Object> parsePreferences(Map<String, ?> allPrefs) throws IOE
         // This only happens for previous usage of setStringSet. The app expects a list.
         List<String> listValue = new ArrayList<>((Set) value);
         // Let's migrate the value too while we are at it.
-        boolean success =
-                preferences
-                        .edit()
-                        .remove(key)
-                        .putString(key, LIST_IDENTIFIER + encodeList(listValue))
-                        .commit();
+        boolean success = preferences
+            .edit()
+            .remove(key)
+            .putString(key, LIST_IDENTIFIER + encodeList(listValue))
+            .commit();
         if (!success) {
-          // If we are unable to migrate the existing preferences, it means we potentially lost them.
-          // In this case, an error from getAllPrefs() is appropriate since it will alert the app during plugin initialization.
+          // If we are unable to migrate the existing preferences, it means we potentially
+          // lost them.
+          // In this case, an error from getAllPrefs() is appropriate since it will alert
+          // the app during plugin initialization.
           throw new IOException("Could not migrate set to list");
         }
         value = listValue;
       }
-      filteredPrefs.put(key, value);
+      if (value != null) {
```

Comment:
that's the only change you should have if you want to do a PR back to the original repo